### PR TITLE
feat(API): add route and implementation for creating/updating repository secret

### PR DESCRIPTION
**Pull request by appleboy**
spec: https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28#create-or-update-a-repository-secret

- Add a new route for creating or updating a secret value in a repository
- Create a new file `routers/api/v1/repo/action.go` with the implementation of the `CreateOrUpdateSecret` function
- Update the Swagger documentation for the `updateRepoSecret` operation in the `v1_json.tmpl` template file

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: routers/api/v1/repo/action.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		}
+		ctx.Status(http.StatusCreated)
+		return
+	}
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "UpdateSecret", err)
+		return
+	}
+
+	ctx.Status(http.StatusNoContent)
```
Comment (wxiaoguang) [inline routers/api/v1/repo/action.go]: Could it share the same code (by a function) as "org"?

Comment (appleboy) [reply routers/api/v1/repo/action.go]: Done. Add new func in model layer

https://github.com/go-gitea/gitea/pull/26766/files#diff-71da4f8401d9a791d2c4b44433b75b4bb6546ae6b17d97dd40aeac727eefd917R164

**Review by wxiaoguang — APPROVED** [review]
Maybe the `CreateOrUpdateSecret` api handler could also be simplified to a shared function (including: check name, call CreateOrUpdateSecret, make response), it also could be done later.

**Comment by puni9869** [comment]
lets roll 💯

**Pull request merged**